tools/codegen/gen.py

```diff
@@ -1001,6 +1004,43 @@ def pre_group_native_functions(
         d[f.func.kind()] = f
     return pre_grouped_native_functions
 
+def get_grouped_by_view_native_functions(
+        native_functions: Sequence[NativeFunction]
+) -> Sequence[Union[NativeFunction, NativeFunctionsViewGroup]]:
+    def maybe_create_view_group(d: Dict[ViewSchemaKind, NativeFunction]) -> List[Union[NativeFunction, NativeFunctionsViewGroup]]:
+        funcs: List[Union[NativeFunction, NativeFunctionsViewGroup]] = []
+        if ViewSchemaKind.aliasing not in d:
+            # Case 1: this op / op group is not aliasing, so we don't create a view group.
+            # return the original (ungrouped) native functions instead.
+            for func in d.values():
+                funcs.append(func)
+        else:
+            # Case 2: this op group contains an aliasing op, so we create a ViewGroup for it.
+            # The handling for out= ops here is unfortunate.
+            # out= ops don't really make sense for view operators.
+            # However, we have at least one existing {view}_copy.out operator in native_functions.yaml.
+            # It shouldn't be part of a view group, so we explicitly don't group it.
+            # There currently aren't any out= view ops (and there probably shouldn't be).
+            # We also expect that when we hit this case, the `non_aliasing` op in the dict
+            # *must* be a view_copy op (this is asserted in the NativeFunctionsViewGroup constructor)
+            if ViewSchemaKind.out in d:
+                funcs.append(d[ViewSchemaKind.out])
+
+            funcs.append(NativeFunctionsViewGroup(
+                view=d[ViewSchemaKind.aliasing],
+                view_copy=d.get(ViewSchemaKind.non_aliasing, None),
+                view_inplace=d.get(ViewSchemaKind.inplace, None),
+            ))
+        return funcs
+
+    grouped_by_views: Dict[FunctionSchema, Dict[ViewSchemaKind, NativeFunction]] = defaultdict(dict)
+    for f in native_functions:
+        schema = f.func.signature(strip_view_copy_name=True)
```

Comment:
nit: it's probably clearer if we have something like `view_signature()` (which can call into an internal helper function; the point is that it is never valid to have both strip kwargs True at the same time)